### PR TITLE
keep JSON format

### PR DESCRIPTION
**Pull request by casperdcl**
updated spec to be more compatible

### CODE REVIEW

File: iterative/utils/analytics.go [hunk 1]
```diff
@@ -212,54 +254,29 @@ func UserId() (string, error) {
 
 	if os.IsNotExist(errorNew) {
 		if !os.IsNotExist(errorOld) {
```
Comment (0x2b3bfa0) [inline iterative/utils/analytics.go:256]: [New code should use `errors.Is` instead](https://pkg.go.dev/os#IsNotExist) 💅🏼 
```suggestion
	if errors.Is(errorNew, fs.ErrNotExist) {
		if !errors.Is(errorOld, fs.ErrNotExist) {
```

**Review by 0x2b3bfa0 — APPROVED** [review]
LMTM :shipit:

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: iterative/utils/analytics.go [hunk 2]
```diff
@@ -212,54 +254,29 @@ func UserId() (string, error) {
...
-
-			defer jsonFile.Close()
 		}
 
 		err := os.MkdirAll(filepath.Dir(new), 0644)
 		if err != nil {
 			return "", err
 		}
-		err = ioutil.WriteFile(new, []byte(id), 0644)
+		err = writeId(new, id)
```
[the remark this answers is not in the document]
Comment (dacbd) [reply iterative/utils/analytics.go:267]: I took another look and the spec document and it does have this as a file as json with: `{"user_id": "id"}`

[the remark this answers is not in the document]
Comment (DavidGOrtega) [reply iterative/utils/analytics.go:268]: so both the same? Then we have it wrong at TPI @dacbd

File: iterative/utils/analytics.go [hunk 3]
```diff
@@ -200,66 +248,41 @@ func UserId() (string, error) {
...
-			}
-			id = data["user_id"].(string)
-
-			defer jsonFile.Close()
 		}
 
 		err := os.MkdirAll(filepath.Dir(new), 0644)
 		if err != nil {
 			return "", err
 		}
```
Comment (DavidGOrtega) [inline iterative/utils/analytics.go:272]:
```suggestion
```

Not needed `writeId` is creating the folder

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: iterative/utils/analytics.go [hunk 4]
```diff
@@ -155,10 +156,57 @@ func GroupId() (string, error) {
...
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(path, bytes, 0644)
 }
 
 func UserId() (string, error) {
+	var id string
+	var err error
```
Comment (DavidGOrtega) [inline iterative/utils/analytics.go:209]: Not sure why do you prefer this 🙂

**Review by DavidGOrtega — APPROVED** [review]

File: iterative/utils/analytics.go (hunk 4 above)
Comment (dacbd) [reply iterative/utils/analytics.go:209]: there are multiple uses of that var at different scopes, IIRC there was a listing error that `:=` was already defined or something.

**Pull request merged**